File: app/src/main/java/com/rdev/tryp/ui_only/screens/help/RecentTripsAdapter.java
package com.rdev.tryp.ui_only.screens.help;

import android.content.Context;
import android.graphics.Typeface;
import android.text.Spannable;
import android.text.SpannableString;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.rdev.tryp.R;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import io.github.inflationx.calligraphy3.CalligraphyTypefaceSpan;
import io.github.inflationx.calligraphy3.TypefaceUtils;

/**
 * Created by Alexey Matrosov on 04.03.2019.
 */
public class RecentTripsAdapter extends RecyclerView.Adapter<RecentTripsAdapter.ViewHolder> {

    private Context context;
    private List<RecentTripItem> items;

    public RecentTripsAdapter(Context context, List<RecentTripItem> items) {
        this.context = context;
        this.items = items;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_recent_trip, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        RecentTripItem item = items.get(position);

        holder.userIcon.setImageResource(item.getUserIcon() == 0 ? R.drawable.invite_default_icon : item.getUserIcon());
        holder.isActive.setVisibility(item.isActive() ? View.VISIBLE : View.INVISIBLE);
        setDate(holder.time, item.getDate());
        holder.message.setText(item.getMessage());
        holder.amount.setText(item.getAmount());
        holder.distance.setText(item.getDistance());
    }

    @Override
    public int getItemCount() {
        return items == null ? 0 : items.size();
    }

    private void setDate(TextView dateView, String date) {
        int index = date.indexOf(", ");
        if (index == -1) {
            dateView.setText(date);
        }

        String boldPart = date.substring(0, index);
        int coloredPart1Index = date.indexOf(boldPart);

        Spannable spannable = new SpannableString(date);
        if (coloredPart1Index != -1 && context != null) {
            Typeface typeface = TypefaceUtils.load(context.getAssets(), "fonts/OpenSans-Bold.ttf");

            spannable.setSpan(new CalligraphyTypefaceSpan(typeface),
                    coloredPart1Index,
                    coloredPart1Index + boldPart.length(),
                    Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
        }

        dateView.setText(spannable, TextView.BufferType.SPANNABLE);
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        ImageView userIcon;
        View isActive;
        TextView time;
        TextView message;
        TextView amount;
        TextView distance;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            userIcon = itemView.findViewById(R.id.recent_user_icon);
            isActive = itemView.findViewById(R.id.recent_active);
            time = itemView.findViewById(R.id.recent_date);
            message = itemView.findViewById(R.id.recent_message);
            amount = itemView.findViewById(R.id.recent_amount);
            distance = itemView.findViewById(R.id.recent_distance);
        }
    }
}

File: app/src/main/java/com/rdev/tryp/blocks/screens/help/RecentTripsAdapter.kt
package com.rdev.tryp.blocks.screens.help

import android.content.Context
import android.text.Spannable
import android.text.SpannableString
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.rdev.tryp.R
import androidx.recyclerview.widget.RecyclerView
import io.github.inflationx.calligraphy3.CalligraphyTypefaceSpan
import io.github.inflationx.calligraphy3.TypefaceUtils

/**
 * Created by Alexey Matrosov on 04.03.2019.
 */


class RecentTripsAdapter(private val context: Context?, private val items: List<RecentTripItem>) : RecyclerView.Adapter<RecentTripsAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
            ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_recent_trip, parent, false))

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]

        holder.userIcon?.setImageResource(if (item.userIcon == 0) R.drawable.invite_default_icon else item.userIcon)
        holder.isActive?.visibility = if (item.isActive) View.VISIBLE else View.INVISIBLE
        setDate(holder.time, item.date)
        holder.message?.text = item.message
        holder.amount?.text = item.amount
        holder.distance?.text = item.distance
    }

    override fun getItemCount() = items.size

    private fun setDate(dateView: TextView?, date: String) {
        val index = date.indexOf(", ")
        if (index == -1) {
            dateView?.text = date
        }

        val boldPart = date.substring(0, index)
        val coloredPart1Index = date.indexOf(boldPart)

        val spannable = SpannableString(date)
        if (coloredPart1Index != -1 && context != null) {
            val typeface = TypefaceUtils.load(context.assets, "fonts/OpenSans-Bold.ttf")

            spannable.setSpan(CalligraphyTypefaceSpan(typeface),
                    coloredPart1Index,
                    coloredPart1Index + boldPart.length,
                    Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        }

        dateView?.setText(spannable, TextView.BufferType.SPANNABLE)
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var userIcon = itemView.findViewById(R.id.recent_user_icon) as? ImageView
        var isActive = itemView.findViewById(R.id.recent_active) as? View
        var time = itemView.findViewById(R.id.recent_date) as? TextView
        var message = itemView.findViewById(R.id.recent_message) as? TextView
        var amount = itemView.findViewById(R.id.recent_amount) as? TextView
        var distance = itemView.findViewById(R.id.recent_distance) as? TextView
    }

}